python_ta/checkers/one_iteration_checker.py

```diff
@@ -53,11 +64,20 @@ def _check_one_iteration(self, node: Union[nodes.For, nodes.While]) -> bool:
         if preds == []:
             return False
 
+        # the loop block is unfeasible
```

Comment:
Overall, I think this algorithm is more complicated than it needs to be. First, you should be able to conditionally filter `preds` directly for only feasible edges after Line 62: `if self.linter.config.z3: preds = ...`

Second, as a general principle whenever you have a comment describing and if statement check, the comment should say something like "Check whether..." rather than stating the condition directly, as that is confusing to the reader.